components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle

```diff
@@ -105,13 +105,14 @@ def registerBuildScanActions = { def buildScan ->
             return System.getProperty(name) ?: System.getenv(envVarName)
         }
         def runNum = getRunNumInputParam('com.gradle.enterprise.build-validation.runNum')
+        def rootProjectName = gradle.rootProject.name
```

Comment:
Previously, the root project name was extracted using `settings.rootProject.name` for Gradle versions >= 6, or from `gradle.rootProject.name` for Gradle versions >= 5. See below.

https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/da175ff75c94e3807ced83167ed779e2dd2d8677/components/scripts/gradle/gradle-init-scripts/capture-published-build-scan.gradle#L15

https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/da175ff75c94e3807ced83167ed779e2dd2d8677/components/scripts/gradle/gradle-init-scripts/capture-published-build-scan.gradle#L22

I think it was probably due to the type of hook being used. In my testing, extracting the root project name via `gradle.rootProject.name` worked in all Gradle versions supported by the scripts (5+), so I did not think it made sense to try and extract it differently based on Gradle version.

Reply:
I just discovered accessing `gradle` from a Groovy closure is not supported by the configuration cache. I will have to rethink this a little.

Reply:
See here: https://github.com/gradle/gradle-enterprise-build-validation-scripts/actions/runs/4692331201/jobs/8317949102#step:8:77

(For some reason it is not captured in the Build Scan)

Reply:
To solve this, I added a `rootProjectName` parameter to the `registerBuildScanActions` function and call either `settings.rootProject.name` or `rootProject.name` depending on Gradle version. 

Both the cross Gradle version tests I ran locally and the [cross platform tests](https://github.com/gradle/gradle-enterprise-build-validation-scripts/actions/runs/4692651896/jobs/8318658237) seemed to like it.

Reply:
Very nice.

Interestingly, it is no also very close to how we pass the root project name in the case of Maven.